index.js

```diff
@@ -0,0 +1,335 @@
+    // gather included styles
+    includedStyles.forEach(ism => {
+      // this style may also have includes
+      inlineStyleIncludes(ism);
+      const inlineDocument = module.__ownerDocument;
+      let includeText = dom5.getTextContent(ism);
+      // adjust paths
+      includeText = pathResolver.rewriteURL(inlineDocument, baseDocument, includeText);
+      styleText.push(includeText);
+    });
+  });
+  // remove inlined includes
+  if (leftover.length) {
+    dom5.setAttribute(style, 'include', leftover.join(' '));
+  } else {
+    dom5.removeAttribute(style, 'include');
+  }
+  // prepend included styles
+  if (styleText.length) {
+    let text = dom5.getTextContent(style);
+    text = styleText.join('') + text;
+    dom5.setTextContent(style, text);
+  }
+}
+
+function applyShim(ast) {
+  /*
+   * `transform` expects an array of decorated <style> elements
+   *
+   * Decorated <style> elements are ones with `__cssRules` property
+   * with a value of the CSS ast
+   */
+  Polymer.StyleUtil.forEachRule(ast, rule => Polymer.ApplyShim.transformRule(rule));
+}
+
+function getModuleDefinition(module, elements) {
+  for (let i = 0; i < elements.length; i++) {
+    const is = elements[i].is;
+    if (is.toLowerCase() === module) {
+      return elements[i];
+    }
+  }
+  return null;
+}
+
+function getTypeExtends(element) {
+  let props = element.properties || [];
+  let ret = '';
+  // loop over element properties with javascript AST
+  for (let i = 0; i < props.length; i++) {
+    if (props[i].name === 'extends') {
+      // node is espree AST node
+      let node = props[i].javascriptNode;
+      // property node has a value node with key and value nodes
+      if (node && node.type === 'Property') {
+        ret = node.value.value;
+        break;
+      }
+    }
+  }
+  return ret.toLowerCase();
+}
+
+function shadyShim(ast, style, elements) {
+  const scope = scopeMap.get(style);
+  const moduleDefinition = getModuleDefinition(scope, elements);
+  // only shim if module is a full polymer element, not just a style module
+  if (!scope || !moduleDefinition) {
+    return;
+  }
+  const ext = getTypeExtends(moduleDefinition);
+  Polymer.StyleTransformer.css(ast, scope, ext);
+}
+
+function addClass(node, className) {
+  const classList = getAttributeArray(node, 'class');
+  classList.push(className, 'style-scope');
+  dom5.setAttribute(node, 'class', classList.join(' '));
+}
+
+function markElement(domModule, scope) {
+  const buildType = Polymer.Settings.useNativeShadow ? 'shadow' : 'shady';
+  dom5.setAttribute(domModule, 'css-build', buildType);
+  // mark elements' subtree under shady build
+  if (buildType === 'shady' && scope) {
+    const template = dom5.query(domModule, pred.hasTagName('template'));
+    // apply scoping to template
+    if (template) {
+      const elements = dom5.queryAll(template, notStyleMatch);
+      elements.forEach(el => addClass(el, scope));
+    }
+  }
+}
+
+module.exports = (paths, options) => {
```

Comment:
I would consider not creating ES-incompatible modules. ie, always export a module-type object.